<?xml version="1.0" encoding="UTF-8"?>
<!-- 
  Example launch file for a costmap fusion node
-->
<launch>
  <arg name="vehicle" default="X1" />
  <arg name="ns" default="/$(arg vehicle)" />

  <node ns="$(arg ns)" pkg="msfm3d" type="costmap_generator" name="costmap_generator" output="screen">
    <param name="fixedFrame" type="string" value="world"/>
    <param name="rate" value="5.0"/>
    <param name="resolution" value="0.2"/>
    <param name="untraversableDistance" value="0.01"/>

    <remap from="esdf" to="voxblox_node/esdf_pointcloud"/>
    <!-- <remap from="octomap_binary" to="octomap_binary"/> -->
    <!-- <remap from="traversability_map" to="traversability_map"/> -->
  </node>

  <!-- <node pkg="nodelet" type="nodelet" name="pcl_manager_costmap" args="manager" output="screen" /> -->

  <!-- Run a passthrough filter to clean NaNs -->
  <node ns="$(arg ns)" pkg="nodelet" type="nodelet" name="passthrough1" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="cost_map" />
    <remap from="~output" to="cost_map_filtered" />
    <rosparam>
      filter_field_name: intensity
      filter_limit_min: 0.2
      filter_limit_max: 100.0
      filter_limit_negative: False
    </rosparam>
  </node>
</launch>
